### PR TITLE
remove duplicated check in operator_generator.go

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

I'm not sure if this check is required for internal client
set, but for versioned clientset, this check is unnecessary becase the fetchErr has been checked.

FYI https://github.com/kubernetes/kubernetes/pull/30737/files#diff-af89d551e56ea6ea4398b82bad790a6228131ad208dc4cbc736a3a62d4b2f404R30

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by carlory** [comment]
/cc @saad-ali 

PTAL

File: pkg/volume/util/operationexecutor/operation_generator.go [hunk 1]
```diff
@@ -1609,13 +1601,6 @@ func (og *operationGenerator) verifyVolumeIsSafeToDetach(
 		return volumeToDetach.GenerateErrorDetailed("DetachVolume failed fetching node from API server", fetchErr)
 	}
 
-	if node == nil {
-		// On failure, return error. Caller will log and retry.
-		return volumeToDetach.GenerateErrorDetailed(
-			"DetachVolume failed fetching node from API server",
-			fmt.Errorf("node object retrieved from API server is nil"))
-	}
-
```
Comment (wangyysde) [inline pkg/volume/util/operationexecutor/operation_generator.go:1618]: Hmm. I agree with you.

**Comment by 21kyu** [comment]
Q: Is it correct to understand that there is no need to check if the response of `Nodes().Get()` which is `*v1.Node` is nil,
ultimately because `storage.Interface.Get` handles the NotFound case?

**Comment by carlory** [comment]
if this function does not return an error, the returned object is non-nil. If the object does not exist, the function will return a notfound type error.

**Pull request merged**